### PR TITLE
[new release] rpclib, rpclib-lwt, rpclib-js, rpclib-html, rpclib-async, rpc and ppx_deriving_rpc (8.1.1)

### PR DESCRIPTION
**Pull request by mseri**
A library to deal with RPCs in OCaml

- Project page: <a href="https://github.com/mirage/ocaml-rpc">https://github.com/mirage/ocaml-rpc</a>
- Documentation: <a href="https://mirage.github.io/ocaml-rpc/rpclib">https://mirage.github.io/ocaml-rpc/rpclib</a>

##### CHANGES:

* Ignore error about using f-strings in python bindings (@psafont)
* Compatibility with rresult 0.7.0 (@psafont)

### CODE REVIEW

File: packages/ppx_deriving_rpc/ppx_deriving_rpc.8.1.1/opam [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+using XML or JSON as transport encodings, and multiple generators
+for documentations, clients, servers, javascript bindings, python
+bindings, ...
+
+The transport mechanism itself is outside the scope of this library
+as all conversions are from and to strings.
+"""
+url {
+  src:
+    "https://github.com/mirage/ocaml-rpc/releases/download/8.1.1/rpclib-8.1.1.tbz"
```
Comment (psafont) [inline packages/ppx_deriving_rpc/ppx_deriving_rpc.8.1.1/opam:34]:
```suggestion
    "https://github.com/mirage/ocaml-rpc/releases/download/v8.1.1/rpclib-8.1.1.tbz"
```

File: packages/rpc/rpc.8.1.1/opam [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+The transport mechanism itself is outside the scope of this library
+as all conversions are from and to strings.
+
+This is a dummy package installing the main library components.
+"""
+post-messages: ["DEPRECATED. This package is a virtual package and is outdated, you should consider using directly rpclib, rpclib-lwt, rpclib-async and ppx_deriving_rpc instead"]
+url {
+  src:
+    "https://github.com/mirage/ocaml-rpc/releases/download/8.1.1/rpclib-8.1.1.tbz"
```
Comment (psafont) [inline packages/rpc/rpc.8.1.1/opam:39]:
```suggestion
    "https://github.com/mirage/ocaml-rpc/releases/download/v8.1.1/rpclib-8.1.1.tbz"
```

File: packages/rpclib-async/rpclib-async.8.1.1/opam [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+using XML or JSON as transport encodings, and multiple generators
+for documentations, clients, servers, javascript bindings, python
+bindings, ...
+
+The transport mechanism itself is outside the scope of this library
+as all conversions are from and to strings.
+"""
+url {
+  src:
+    "https://github.com/mirage/ocaml-rpc/releases/download/8.1.1/rpclib-8.1.1.tbz"
```
Comment (psafont) [inline packages/rpclib-async/rpclib-async.8.1.1/opam:33]:
```suggestion
    "https://github.com/mirage/ocaml-rpc/releases/download/v8.1.1/rpclib-8.1.1.tbz"
```

File: packages/rpclib-html/rpclib-html.8.1.1/opam [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+using XML or JSON as transport encodings, and multiple generators
+for documentations, clients, servers, javascript bindings, python
+bindings, ...
+
+The transport mechanism itself is outside the scope of this library
+as all conversions are from and to strings.
+"""
+url {
+  src:
+    "https://github.com/mirage/ocaml-rpc/releases/download/8.1.1/rpclib-8.1.1.tbz"
```
Comment (psafont) [inline packages/rpclib-html/rpclib-html.8.1.1/opam:29]:
```suggestion
    "https://github.com/mirage/ocaml-rpc/releases/download/v8.1.1/rpclib-8.1.1.tbz"
```

File: packages/rpclib-js/rpclib-js.8.1.1/opam [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+using XML or JSON as transport encodings, and multiple generators
+for documentations, clients, servers, javascript bindings, python
+bindings, ...
+
+The transport mechanism itself is outside the scope of this library
+as all conversions are from and to strings.
+"""
+url {
+  src:
+    "https://github.com/mirage/ocaml-rpc/releases/download/8.1.1/rpclib-8.1.1.tbz"
```
Comment (psafont) [inline packages/rpclib-js/rpclib-js.8.1.1/opam:30]:
```suggestion
    "https://github.com/mirage/ocaml-rpc/releases/download/v8.1.1/rpclib-8.1.1.tbz"
```

File: packages/rpclib-lwt/rpclib-lwt.8.1.1/opam [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+using XML or JSON as transport encodings, and multiple generators
+for documentations, clients, servers, javascript bindings, python
+bindings, ...
+
+The transport mechanism itself is outside the scope of this library
+as all conversions are from and to strings.
+"""
+url {
+  src:
+    "https://github.com/mirage/ocaml-rpc/releases/download/8.1.1/rpclib-8.1.1.tbz"
```
Comment (psafont) [inline packages/rpclib-lwt/rpclib-lwt.8.1.1/opam:34]:
```suggestion
    "https://github.com/mirage/ocaml-rpc/releases/download/v8.1.1/rpclib-8.1.1.tbz"
```

**Review by psafont — COMMENTED** [review]
the base rpclib uses a different url from the rest of packages. Github tends to return 404 for the url that the other packages use. I suggest to use the same URL for all

**Comment by mseri** [comment]
Oh yes, that is a bug in dune-release, thanks for the fix

File: packages/rpclib/rpclib.8.1.1/opam [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+homepage: "https://github.com/mirage/ocaml-rpc"
+doc: "https://mirage.github.io/ocaml-rpc/rpclib"
+bug-reports: "https://github.com/mirage/ocaml-rpc/issues"
+depends: [
+  "ocaml" {>= "4.04.0"}
+  "alcotest" {with-test}
+  "dune" {>= "2.0.0"}
+  "base64" {>= "3.4.0"}
+  "cmdliner" {>= "0.9.8"}
+  "rresult"
```
Comment (mseri) [inline packages/rpclib/rpclib.8.1.1/opam:15]:
```suggestion
  "rresult" {>= "0.3.0"}
```
Rpclib somehow relies on the Result module and the result compatibility shim being present

File: packages/ppx_deriving_rpc/ppx_deriving_rpc.8.1.1/opam [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+authors: "Thomas Gazagnaire, Jon Ludlam"
+tags: ["org:mirage" "org:xapi-project"]
+homepage: "https://github.com/mirage/ocaml-rpc"
+doc: "https://mirage.github.io/ocaml-rpc/ppx_deriving_rpc"
+bug-reports: "https://github.com/mirage/ocaml-rpc/issues"
+depends: [
+  "ocaml" {>= "4.08"}
+  "dune" {>= "2.0.0"}
+  "rpclib" {= version}
+  "rresult"
```
Comment (mseri) [inline packages/ppx_deriving_rpc/ppx_deriving_rpc.8.1.1/opam:13]:
```suggestion
  "rresult" {>= "0.3.0"}
```

File: packages/rpclib-html/rpclib-html.8.1.1/opam [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+authors: "Thomas Gazagnaire, Jon Ludlam"
+tags: ["org:mirage" "org:xapi-project"]
+homepage: "https://github.com/mirage/ocaml-rpc"
+doc: "https://mirage.github.io/ocaml-rpc/rpclib-html"
+bug-reports: "https://github.com/mirage/ocaml-rpc/issues"
+depends: [
+  "ocaml"
+  "dune"
+  "rpclib" {=version}
+  "cow"
```
Comment (mseri) [inline packages/rpclib-html/rpclib-html.8.1.1/opam:14]:
```suggestion
  "cow" {>= "2.0.0"}
```

File: packages/ppx_deriving_rpc/ppx_deriving_rpc.8.1.1/opam [hunk 10]
```diff
@@ -0,0 +1,40 @@
+opam-version: "2.0"
+synopsis: "Ppx deriver for ocaml-rpc, a library to deal with RPCs in OCaml"
+maintainer: "thomas@gazagnaire.org"
+authors: "Thomas Gazagnaire, Jon Ludlam"
+tags: ["org:mirage" "org:xapi-project"]
+homepage: "https://github.com/mirage/ocaml-rpc"
+doc: "https://mirage.github.io/ocaml-rpc/ppx_deriving_rpc"
+bug-reports: "https://github.com/mirage/ocaml-rpc/issues"
+depends: [
```
Comment (mseri) [inline packages/ppx_deriving_rpc/ppx_deriving_rpc.8.1.1/opam:9]:
```suggestion
license: "ISC"
depends: [
```

File: packages/rpc/rpc.8.1.1/opam [hunk 11]
```diff
@@ -0,0 +1,45 @@
+opam-version: "2.0"
+synopsis: "A library to deal with RPCs in OCaml - meta-package"
+maintainer: "thomas@gazagnaire.org"
+authors: "Thomas Gazagnaire, Jon Ludlam"
+tags: ["org:mirage" "org:xapi-project" "deprecated"]
+homepage: "https://github.com/mirage/ocaml-rpc"
+doc: "https://mirage.github.io/ocaml-rpc/rpc"
+bug-reports: "https://github.com/mirage/ocaml-rpc/issues"
```
Comment (mseri) [inline packages/rpc/rpc.8.1.1/opam:8]:
```suggestion
license: "ISC"
depends: [
```

File: packages/rpc/rpc.8.1.1/opam [hunk 12]
```diff
@@ -0,0 +1,45 @@
...
+depends: [
```
Comment (mseri) [inline packages/rpc/rpc.8.1.1/opam:9]:
```suggestion
depends: [
```

File: packages/rpclib-async/rpclib-async.8.1.1/opam [hunk 13]
```diff
@@ -0,0 +1,39 @@
+opam-version: "2.0"
+synopsis: "A library to deal with RPCs in OCaml - Async interface"
+maintainer: "thomas@gazagnaire.org"
+authors: "Thomas Gazagnaire, Jon Ludlam"
+tags: ["org:mirage" "org:xapi-project"]
+homepage: "https://github.com/mirage/ocaml-rpc"
+doc: "https://mirage.github.io/ocaml-rpc/rpclib-async"
+bug-reports: "https://github.com/mirage/ocaml-rpc/issues"
+depends: [
```
Comment (mseri) [inline packages/rpclib-async/rpclib-async.8.1.1/opam:9]:
```suggestion
license: "ISC"
depends: [
```

File: packages/rpclib-html/rpclib-html.8.1.1/opam [hunk 14]
```diff
@@ -0,0 +1,35 @@
+opam-version: "2.0"
+synopsis:
+  "A library to deal with RPCs in OCaml - html documentation generator"
+maintainer: "thomas@gazagnaire.org"
+authors: "Thomas Gazagnaire, Jon Ludlam"
+tags: ["org:mirage" "org:xapi-project"]
+homepage: "https://github.com/mirage/ocaml-rpc"
+doc: "https://mirage.github.io/ocaml-rpc/rpclib-html"
+bug-reports: "https://github.com/mirage/ocaml-rpc/issues"
+depends: [
```
Comment (mseri) [inline packages/rpclib-html/rpclib-html.8.1.1/opam:10]:
```suggestion
license: "ISC"
depends: [
```

File: packages/rpclib-js/rpclib-js.8.1.1/opam [hunk 15]
```diff
@@ -0,0 +1,36 @@
+opam-version: "2.0"
+synopsis: "A library to deal with RPCs in OCaml - Bindings for js_of_ocaml"
+maintainer: "thomas@gazagnaire.org"
+authors: "Thomas Gazagnaire, Jon Ludlam"
+tags: ["org:mirage" "org:xapi-project"]
+homepage: "https://github.com/mirage/ocaml-rpc"
+doc: "https://mirage.github.io/ocaml-rpc/rpclib-js"
+bug-reports: "https://github.com/mirage/ocaml-rpc/issues"
+depends: [
```
Comment (mseri) [inline packages/rpclib-js/rpclib-js.8.1.1/opam:9]:
```suggestion
license: "ISC"
depends: [
```

File: packages/rpclib-lwt/rpclib-lwt.8.1.1/opam [hunk 16]
```diff
@@ -0,0 +1,40 @@
+opam-version: "2.0"
+synopsis: "A library to deal with RPCs in OCaml - Lwt interface"
+maintainer: "thomas@gazagnaire.org"
+authors: "Thomas Gazagnaire, Jon Ludlam"
+tags: ["org:mirage" "org:xapi-project"]
+homepage: "https://github.com/mirage/ocaml-rpc"
+doc: "https://mirage.github.io/ocaml-rpc/rpclib-lwt"
+bug-reports: "https://github.com/mirage/ocaml-rpc/issues"
+depends: [
```
Comment (mseri) [inline packages/rpclib-lwt/rpclib-lwt.8.1.1/opam:9]:
```suggestion
license: "ISC"
depends: [
```

File: packages/rpclib/rpclib.8.1.1/opam [hunk 17]
```diff
@@ -0,0 +1,44 @@
+opam-version: "2.0"
+synopsis: "A library to deal with RPCs in OCaml"
+maintainer: "thomas@gazagnaire.org"
+authors: "Thomas Gazagnaire, Jon Ludlam"
+tags: ["org:mirage" "org:xapi-project"]
+homepage: "https://github.com/mirage/ocaml-rpc"
+doc: "https://mirage.github.io/ocaml-rpc/rpclib"
+bug-reports: "https://github.com/mirage/ocaml-rpc/issues"
+depends: [
```
Comment (mseri) [inline packages/rpclib/rpclib.8.1.1/opam:9]:
```suggestion
license: "ISC"
depends: [
```

File: packages/rpclib-lwt/rpclib-lwt.8.1.1/opam [hunk 18]
```diff
@@ -0,0 +1,40 @@
...
+  "ocaml"
+  "alcotest" {with-test}
+  "dune"
```
Comment (mseri) [inline packages/rpclib-lwt/rpclib-lwt.8.1.1/opam:12]:
```suggestion
  "dune" {>= "2.0.0"}
```

File: packages/rpclib-js/rpclib-js.8.1.1/opam [hunk 19]
```diff
@@ -0,0 +1,36 @@
...
+  "ocaml"
+  "dune"
```
Comment (mseri) [inline packages/rpclib-js/rpclib-js.8.1.1/opam:11]:
```suggestion
  "dune" {>= "2.0.0"}
```

File: packages/rpclib-html/rpclib-html.8.1.1/opam [hunk 20]
```diff
@@ -0,0 +1,35 @@
...
+  "ocaml"
+  "dune"
```
Comment (mseri) [inline packages/rpclib-html/rpclib-html.8.1.1/opam:12]:
```suggestion
  "dune" {>= "2.0.0"}
```

File: packages/rpclib-async/rpclib-async.8.1.1/opam [hunk 21]
```diff
@@ -0,0 +1,39 @@
...
+  "ocaml"
+  "alcotest" {with-test}
+  "dune"
```
Comment (mseri) [inline packages/rpclib-async/rpclib-async.8.1.1/opam:12]:
```suggestion
  "dune" {>= "2.0.0"}
```

File: packages/rpc/rpc.8.1.1/opam [hunk 22]
```diff
@@ -0,0 +1,45 @@
...
+  "ocaml" {>= "4.04.1"}
+  "dune"
```
Comment (mseri) [inline packages/rpc/rpc.8.1.1/opam:11]:
```suggestion
  "dune" {>= "2.0.0"}
```

**Comment by mseri** [comment]
Woof, lots of upstreaming to do…

File: packages/rpc/rpc.8.1.1/opam [hunk 23]
```diff
@@ -0,0 +1,46 @@
...
+  "rpclib-lwt" {=version}
+  "ppx_deriving_rpc" {=version}
+  "alcotest" {with-test}
+  "md2mld" {with-doc & >= "0.4.0"}
+  "conf-which" {with-test}
+  "odoc" {with-doc}
+]
+build: [
+    ["dune" "runtest" "-p" name] {with-test}
+    ["dune" "build" "@doc"] {with-doc}
```
Comment (kit-ty-kate) [inline packages/rpc/rpc.8.1.1/opam:23]:
```suggestion
    ["dune" "build" "@doc" "-p" name "-j" jobs] {with-doc}
```

File: packages/rpclib/rpclib.8.1.1/opam [hunk 24]
```diff
@@ -0,0 +1,45 @@
...
+  "rresult" {>= "0.3.0"}
+  "xmlm"
+  "yojson" {>= "1.7.0"}
+]
+conflicts: [
+  "result" {< "1.5"}
+]
+build: [
+  ["dune" "build" "-p" name "-j" jobs]
+  ["dune" "runtest" "-p" name] {with-test}
```
Comment (kit-ty-kate) [inline packages/rpclib/rpclib.8.1.1/opam:25]:
```suggestion
  ["dune" "runtest" "-p" name "-j" jobs] {with-test}
```

File: packages/rpclib-lwt/rpclib-lwt.8.1.1/opam [hunk 25]
```diff
@@ -0,0 +1,41 @@
...
+  "alcotest" {with-test}
+  "dune" {>= "2.0.0"}
+  "rpclib" {=version}
+  "lwt" {>= "3.0.0"}
+  "alcotest-lwt" {with-test}
+  "ppx_deriving_rpc" {with-test & =version}
+]
+build: [
+  ["dune" "build" "-p" name "-j" jobs]
+  ["dune" "runtest" "-p" name] {with-test}
```
Comment (kit-ty-kate) [inline packages/rpclib-lwt/rpclib-lwt.8.1.1/opam:21]:
```suggestion
  ["dune" "runtest" "-p" name "-j" jobs] {with-test}
```

File: packages/rpclib-async/rpclib-async.8.1.1/opam [hunk 26]
```diff
@@ -0,0 +1,40 @@
...
+  "ocaml"
+  "alcotest" {with-test}
+  "dune" {>= "2.0.0"}
+  "rpclib" {=version}
+  "async" {>= "v0.9.0"}
+  "ppx_deriving_rpc" {with-test & =version}
+]
+build: [
+  ["dune" "build" "-p" name "-j" jobs]
+  ["dune" "runtest" "-p" name] {with-test}
```
Comment (kit-ty-kate) [inline packages/rpclib-async/rpclib-async.8.1.1/opam:20]:
```suggestion
  ["dune" "runtest" "-p" name "-j" jobs] {with-test}
```

File: packages/rpc/rpc.8.1.1/opam [hunk 27]
```diff
@@ -0,0 +1,46 @@
...
+  "rpclib" {=version}
+  "rpclib-lwt" {=version}
+  "ppx_deriving_rpc" {=version}
+  "alcotest" {with-test}
+  "md2mld" {with-doc & >= "0.4.0"}
+  "conf-which" {with-test}
+  "odoc" {with-doc}
+]
+build: [
+    ["dune" "runtest" "-p" name] {with-test}
```
Comment (kit-ty-kate) [inline packages/rpc/rpc.8.1.1/opam:22]:
```suggestion
    ["dune" "runtest" "-p" name "-j" jobs] {with-test}
```

**Comment by kit-ty-kate** [comment]
Thanks!

**Pull request merged**